### DISCUSSION TITLE
Assistance with Battery Modeling for Ancillary Services

### DISCUSSION THREAD

**Discussion by zash-se** [pybamm-team/PyBaMM#4554] [General]
Hello,

I would like to seek your assistance regarding the modeling of batteries for ancillary services, specifically for FFR, FCR-D, and FCR-N:
- FFR (Fast Frequency Response): A service where batteries or other energy sources respond rapidly to deviations in grid frequency. The primary goal is to stabilize the frequency by either injecting or absorbing power almost instantaneously (typically within 1 second).
- FCR-D (Frequency Containment Reserve - Disturbance): A service used to maintain the grid's frequency during large disturbances, such as sudden imbalances between supply and demand. It requires quick ramping of power in response to frequency deviations beyond a certain threshold.
- FCR-N (Frequency Containment Reserve - Normal): A service for maintaining the grid's frequency during normal operations by constantly balancing small deviations in supply and demand. FCR-N operates over longer timescales than FFR but is essential for maintaining overall frequency stability.
To provide a broader context, the real-time grid frequency information can be observed here: [SVK Control Room](https://www.svk.se/en/national-grid/the-control-room/), and data about FCR services can be accessed here: [FCR Data](https://mimer.svk.se/PrimaryRegulation/PrimaryRegulationIndex).
As I do not have a background in battery technology, I find myself a bit lost and would greatly appreciate your help.

My objective, in short, is to model a battery system capable of participating in these ancillary services.
The problem:
I encountered the following experimental data example:
```python
exp = pybamm.Experiment(
   [
       "Hold at 4.2 V until C/100 (5 minute period)",
       "Rest for 4 hours (5 minute period)",
       "Discharge at 0.1C until 2.5 V (5 minute period)", # initial capacity check
       "Charge at 0.3C until 4.2 V (5 minute period)",
       "Hold at 4.2 V until C/100 (5 minute period)",
   ]
)
```
My question is: How should I determine the appropriate charge and discharge rates, the cut-off voltage levels, and the duration for the rest periods?

Additional considerations:
- The model should include battery degradation over time.
- I should be able to determine both the capacity and size of the battery. Specifically, I would like to distinguish between grid-scale batteries and EV batteries. My focus is on both first and second-life grid-scale batteries, as well as second-life EV batteries.

 Any advice, direction, or guidance would be greatly appreciated.

Thank you very much in advance for your help. I look forward to your response.
Kind regards,

**Comment by brosaplanella** [comment]
If I understand correctly, I think [liionpack](https://github.com/pybamm-team/liionpack) might help you, as it allows you to combine several PyBaMM models which could be used to represent the grid. As we are a community driven project, we cannot provide one-on-one support directly, but maybe some other people in the community can help you and start a collaboration.
